btcstaking-tracker/btcslasher/slasher.go

```diff
@@ -246,13 +246,18 @@ func (bs *BTCSlasher) SlashFinalityProvider(extractedfpBTCSK *btcec.PrivateKey)
 		return fmt.Errorf("failed to get BTC delegations under finality provider %s: %w", fpBTCPK.MarshalHex(), err)
 	}
 
+	// Initialize a mutex to protect access to extractedfpBTCSK *btcec.PrivateKey
+	var mu sync.Mutex
```

Comment:
if we need the mutex to protect access, maybe it is better to remove the wait group entirely 
or wrap the `extractedfpBTCSK *secp256k1.PrivateKey` into a protected structure